test/ConfigAggregatorTest.php

```diff
@@ -92,10 +92,61 @@ function () {
         ], $this->cacheFile);
         $this->assertTrue(file_exists($this->cacheFile));
         $cachedConfig = include $this->cacheFile;
+
         $this->assertInternalType('array', $cachedConfig);
         $this->assertEquals(['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true], $cachedConfig);
     }
 
+    public function testConfigAggregatorCanCacheConfigWithClosures()
+    {
+        if (! is_callable("Brick\VarExporter\VarExporter::export")) {
+            $this->markTestSkipped(
+                'Brick\VarExporter is not available.'
+            );
+        }
+
+        new ConfigAggregator([
+                function () {
+                    return [
+                     'toUpper' => function ($input) {
+                        return strtoupper($input);
+                     },
+                     ConfigAggregator::ENABLE_CACHE => true
+                    ];
+                }
+         ], $this->cacheFile);
+        $this->assertTrue(file_exists($this->cacheFile));
+
+        $cachedConfig = include $this->cacheFile;
+
+        $this->assertTrue(is_callable($cachedConfig['toUpper']));
+        $this->assertEquals('FOOBAR', call_user_func($cachedConfig['toUpper'], 'foobar'));
+    }
+
+    public function testConfigAggregatorRaisesExceptionIfConfigCannotBeExported()
+    {
+        if (! is_callable("Brick\VarExporter\VarExporter::export")) {
+            $this->markTestSkipped(
+                'Brick\VarExporter is not available.'
+            );
+        }
+
+        $this->expectException(ConfigCannotBeCachedException::class);
```

Comment:
This is actually an upside of using `Brick\VarExporter`.
The exception is thrown early during cache generation and not during the second request when the (unusable) cache file is read.